src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java

```diff
@@ -257,54 +257,14 @@ private BuildConfigProjectProperty buildConfigProjectForJob(WorkflowJob job) {
-  then sync plugin is not able to find buildconfig by repo name while
-  syncing back to openshift after creating a buildconfig with strategy
-  JenkinsPipeline, and create a new buildconfig repo name, To fix this
-  issue, added a label of gitRepository name so it will first try to
-  find the BuildConfig with name of jenkins job, if that is not present
-  that is our case then it will also try to find the buildconfig having
-  label with the name of jenkins job and then if it is not present will
-  create a new BuildConfig*/
-
   private void upsertBuildConfigForJob(WorkflowJob job, BuildConfigProjectProperty buildConfigProjectProperty) {
     boolean create = false;
 
-    logger.info("Finding BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() + " name: " +
-      buildConfigProjectProperty.getName());
-
-    BuildConfig jobBuildConfig = getOpenShiftClient().buildConfigs().
-            inNamespace(buildConfigProjectProperty.getNamespace()).withName(buildConfigProjectProperty.getName()).get();
-
-    if (jobBuildConfig == null){
-
-      logger.info("Not able to find BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() + " name: " +
-        buildConfigProjectProperty.getName());
-
-      logger.info("Finding BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() +
-        " with label gitRepository: " + buildConfigProjectProperty.getName());
-
-      BuildConfigList jobBuildConfigs = getOpenShiftClient().buildConfigs().
-        inNamespace(buildConfigProjectProperty.getNamespace())
-        .withLabels(Collections.singletonMap(OPENSHIFT_LABELS_BUILD_CONFIG_GIT_REPOSITORY_NAME, buildConfigProjectProperty.getName())).list();
-
-      /*Always choose the first one because launcher(https://github.com/fabric8-launcher/launcher-backend) will create
-       a single pipeline for a git repo and this will always return one if exists and nothing if does not exist*/
-      if (!jobBuildConfigs.getItems().isEmpty()){
-        jobBuildConfig = jobBuildConfigs.getItems().get(0);
-        logger.info("Able to find BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() +
-          " with label gitRepository: " + buildConfigProjectProperty.getName());
-      } else {
-        logger.info("Not able to find BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() +
-          " with label gitRepository: " + buildConfigProjectProperty.getName());
-      }
-    } else {
-      logger.info("Able to find BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() + " name: " +
-        buildConfigProjectProperty.getName());
-    }
+    //Find the buildConfig if present with both name and label
+    BuildConfig jobBuildConfig = OpenShiftUtils.getBuildConfig(buildConfigProjectProperty.getNamespace(),buildConfigProjectProperty.getName());
 
     if (jobBuildConfig == null) {
-
+      /* Lets create a new buildConfig*/
```

Comment:
Since you are in this path, it might be worth it for you all to port the concurrency fix we have for trying to create a BC when a delete (presumably from the OpenShift side) is in progress.

See https://github.com/openshift/jenkins-sync-plugin/blob/master/src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java#L170 and surrounding code for reference.

Reply:
Thanks for this fix @gabemontero :+1: